modules/matchPattern.js

```diff
@@ -23,7 +23,7 @@ const getMatcher = (pattern, exactly) => {
 
 const parseParams = (pattern, match, keys) =>
   match.slice(1).reduce((params, value, index) => {
```

Comment:
we may want to add `filter(value => value !== undefined)` before the `reduce` call to prevent "missing" params to be defined as param properties at all.